lib/macos.c

```diff
@@ -24,9 +24,8 @@
 
 #include "curl_setup.h"
 
-#if defined(__APPLE__)
-
-#if !defined(TARGET_OS_OSX) || TARGET_OS_OSX
+#if defined(__APPLE__) && TARGET_OS_MAC && \
+  !(defined(TARGET_OS_IPHONE) && TARGET_OS_IPHONE)
```

Comment:
Can't this be just `#ifdef CURL_OSX_CALL_COPYPROXIES` now?

Reply:
Two things about that.

First, I think the intention was that `Curl_macos_init` would always be a real function on macOS and a macro that does nothing on iOS and every other OS. Currently, that real function calls `SCDynamicStoreCopyProxies` in some configurations and does nothing in others. So yes, I think that for the way this currently works it could be changed so that `Curl_macos_init` would only be a real function when `SCDynamicStoreCopyProxies` needs to be called and a macro that does nothing otherwise. But I think the reason it was set up this way is because at some time in the future you might want to have some other macOS-specific init code and if we change the condition under which `Curl_macos_init` is a real function now, that would have to be reverted at such a future time.

Second, everywhere that `CURL_OSX_CALL_COPYPROXIES` is referenced at present it is done this way:
```c
#if defined(ENABLE_IPV6) && defined(CURL_OSX_CALL_COPYPROXIES)
```
It seems like it would be better to define `CURL_OSX_CALL_COPYPROXIES` only if `ENABLE_IPV6` is also defined. Do you know any reason why that wasn't done?

Reply:
> I think the intention was that Curl_macos_init would always be a real function on macOS

I think no matter what the intention was, if there is nothing to do in the function then it doesn't need to exist. In fact, having an empty function just causes compiler warnings at times. I think it should only exist when it actually has something to do.

> at some time in the future you might want to...

If that ever happens, we can change and improve the logic *then*.

> It seems like it would be better to define CURL_OSX_CALL_COPYPROXIES only if ENABLE_IPV6 is also defined.

I completely agree!

> Do you know any reason why that wasn't done?

Probably just an oversight.